src/ring_theory/polynomial/basic.lean

```diff
@@ -596,3 +633,19 @@ instance {R : Type u} {σ : Type v} [integral_domain R] :
   .. (by apply_instance : comm_ring (mv_polynomial σ R)) }
 
 end mv_polynomial
+
+namespace polynomial
+open unique_factorization_monoid
+
+variables {D : Type u} [integral_domain D] [unique_factorization_monoid D]
+
+protected theorem unique_factorization_monoid : unique_factorization_monoid (polynomial D) :=
+begin
+  haveI := arbitrary (normalization_monoid D),
+  haveI := to_gcd_monoid D,
+  exact ufm_of_gcd_of_wf_dvd_monoid
+end
+
+attribute [priority 100] [instance] polynomial.unique_factorization_monoid
```

Comment:
I'm confused. Why the `protected theorem` and then this global attribute? Can't you make it an `instance` directly?

Reply:
I'm just copying what they did for the Hilbert Basis theorem, also on this file.
`protected theorem` throws an error, so it may be the only way to decide what the instance name is and still keep it protected to a namespace.

Reply:
Anyway, I've reverted my `instance`s to how I originally would have done them, although I haven't changed Hilbert's basis theorem.